app/models/conversation.rb

```diff
@@ -4,20 +4,61 @@ class Conversation < ApplicationRecord
   has_many :messages
   belongs_to :ride
 
-  before_save :update_lifecycle
   before_save :check_ride_attached
+  before_save :update_status_and_lifecycle
   after_create :notify_creation
   around_save :notify_update
 
-  enum status: { in_progress: 0, ride_created: 1, closed: 2 }
-  enum lifecycle: { need_language: 100, need_name: 200, need_origin: 300, need_destination: 400, need_time: 500, info_complete: 1000 }
+  include HasAddress
+
+  enum status: { sms_created: -1, in_progress: 0, ride_created: 1, closed: 2, help_needed: 3 }
+  enum lifecycle: {
```

Comment:
We should probably create a wiki page documenting the bot conversation lifecycle and flows--this is the UX for end users. Documentation will let non-technical contributors understand what's happening, copyedit, etc.